### ISSUE TITLE
Ukrainian callsigns

### ISSUE THREAD

**Issue by diygoodies** [camilstaps/ReverseBeaconClient#10]
Hello you made nice application but there is one problem. Ukrainian callsigns marked as europian russia and have russian flag instead Ukrainian as it realy should. Here is a Ukrainian callsign map http://hfdx.at.ua/img/ua_map.gif

**Comment by camilstaps** [comment]
Thanks. I'm using callsign prefix lists from elsewhere. Could you indicate which callsigns, for example, are listed incorrectly? That will make it easier to debug the issue.

**Comment by camilstaps** [comment]
Closing due to lack of information. Feel free to reopen when more information is available.

**Comment by diygoodies** [comment]
Hi I have send you a link to map with range of ukrainian callsigns.http://hfdx.at.ua/img/ua_map.gif

**Comment by diygoodies** [comment]
UR-UY, EM-EO, UV, UW, UZ, U5 that's how ukrainian callsign starts with

**Comment by camilstaps** [comment]
Looks like this is because Russia has the prefix `U` in the [cty.dat](http://www.country-files.com/cty-dat-format/); this is overridden by the Ukrainian prefixes `U5,UR,...` but the current system does not take that into account.

I'll delay this issue to the point where a more intelligent intermediate representation is used. Currently, the callsign table is a simple list and looking up the country requires going through that list. That can be made much faster with a search tree; a representation that the Python tool can already put in an easy-to-parse txt file that would be read by the app on start-up.

TL;DR: thanks for this, I'll get working on it, but it may take some time.

**Comment by camilstaps** [comment]
This has been fixed in https://github.com/camilstaps/ReverseBeaconClient/commit/c535bd8e7f1deee0607cdf979d389eb1c42b1a86. A new version, v1.2.0, has been published in the Play store (it may take a few hours for it to appear).
